components/Chat/ChatMessage.tsx

```diff
@@ -2,12 +2,12 @@ import { Message } from '@/types/chat';
 import { IconEdit } from '@tabler/icons-react';
 import { useTranslation } from 'next-i18next';
 import { FC, memo, useEffect, useRef, useState } from 'react';
+import { IconCheck, IconCopy } from '@tabler/icons-react';
 import rehypeMathjax from 'rehype-mathjax';
 import remarkGfm from 'remark-gfm';
 import remarkMath from 'remark-math';
 import { CodeBlock } from '../Markdown/CodeBlock';
 import { MemoizedReactMarkdown } from '../Markdown/MemoizedReactMarkdown';
-import { CopyButton } from './CopyButton';
```

Comment:
heads up to @thejackw / @mckaywrigley 

deleting & reintegrating the CopyButton here for 2 reasons:
1) There are plenty of buttons in the UI - this button is not doing anything particularly more complicated than others. That said, I do think there is a case for refactoring around buttons, but I believe we could focus on DRY principles to reduce duplication  
1) The existing approach for visualizing this (and other buttons) relied on the user hovering over a parent area to see the icon. <br><br>When not hovered, the button was completely removed from the DOM. This makes tab-navigation **impossible**. <br><br>I refactored so that we instead put the icons way way offscreen (`translate-x-[1000px]`) and then pop them back to `translate-x-0` when we tab through. For hover, we want the icon to appear when we hover over the parent. I achieved this using tailwind visibility extension<br><br>Once this was done, it meant some the classnames on the CopyButton.jsx component (the translate and group styles) were tightly-coupled to the Chatmessage.jsx, so it really no longer made sense to keep it as a separate component.